scene/resources/primitive_meshes.cpp

```diff
@@ -789,7 +789,7 @@ void CylinderMesh::create_mesh_array(Array &p_arr, float top_radius, float botto
 
 			Vector3 p = Vector3(x * radius, y, z * radius);
 			points.push_back(p);
-			normals.push_back(Vector3(x, 0.0, z));
+			normals.push_back(Vector3(x, (bottom_radius - top_radius)/height, z).normalized());
 			ADD_TANGENT(z, 0.0, -x, 1.0)
```

Comment:
Isn't the tangent wrong too?

Reply:
As you pointed out this is likely the tangent along the circular section and that doesn't change if the mesh is a cone.